### ISSUE TITLE
aria-required-children's reviewEmpty option is broken on hidden children

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4010]
The following should be reported as incomplete of `aria-required-children`, rather than as a violation:

```html
<div role="menu">
  <a href="#" role="menuitem" hidden>foo</a>
</div>
```

This bug was introduced in https://github.com/dequelabs/axe-core/pull/3949, which told axe to ignore the roles of hidden children, but doesn't ignore hidden children when checking `reviewEmpty`.
